composer/algorithms/blurpool/blurpool_layers.py

```diff
@@ -54,24 +61,31 @@ def blur_2d(input: torch.Tensor, stride: _size_2_t = 1, filter: Optional[torch.T
     Returns:
         The blurred input
     """
-    _, c, h, w = input.shape
-    n_in_channels = c
 
     if filter is None:
         filter = _default_2d_filter()
-    if (filter.shape[0] == 1) and (n_in_channels > 1):
-        # filt is already a rank 4 tensor
-        filter = filter.repeat((n_in_channels, 1, 1, 1))
-
-    _, _, filter_h, filter_w = filter.shape
+    # The dynamic control flow branch below does not affect the padding as only h and w are used.
     padding = _padding_for_filt_2d_same(filter)
 
-    if h + 2 * padding[0] < filter_h:
-        return input
-    if w + 2 * padding[1] < filter_w:
-        return input
+    if channels < 1:  # Use Dynamic Control Flow
+        _, c, h, w = input.shape
+        n_in_channels = c
+
+        if (filter.shape[0] == 1) and (n_in_channels > 1):
+            # filt is already a rank 4 tensor
+            filter = filter.repeat((n_in_channels, 1, 1, 1))
+
+        _, _, filter_h, filter_w = filter.shape
+        if h + 2 * padding[0] < filter_h:
+            return input
+        if w + 2 * padding[1] < filter_w:
+            return input
+
+        # Use Dynamic Control Flow
+        return F.conv2d(input, filter, stride=stride, padding=padding, groups=n_in_channels, bias=None)
```

Comment:
```suggestion
```
saves a few loc and slightly clearer if we just set `channels` dynamically and have a shared call to conv2d